examples/ai-core/src/generate-text/openai-multimodal-url-download.ts

```diff
@@ -0,0 +1,34 @@
+import { openai } from '@ai-sdk/openai';
+import { generateText } from 'ai';
+import 'dotenv/config';
+
+async function main() {
+  const result = await generateText({
+    model: openai('gpt-4o', {
+      // AI SDK will download the images and add them as data:
+      downloadImages: true,
+    }),
+    messages: [
+      {
+        role: 'user',
+        content: [
+          { type: 'text', text: 'Describe the image in detail.' },
+          {
+            type: 'image',
+            image:
+              'https://github.com/vercel/ai/blob/main/examples/ai-core/data/comic-cat.png?raw=true',
+
+            // OpenAI specific extension: image detail:
+            experimental_providerMetadata: {
+              openai: { detail: 'low' },
```

Comment:
question – do we enforce the types so that `detail` is only allowed for image message parts?

Reply:
no - provider extensions have no type enforcement

Reply:
you can set any json (it's an escape hatch that comes with this downside)

Reply:
openai calls it `detail` and i want to be in sync with that since it's part of image (considered `imageDetail`)

Reply:
changing it to `imageDetail` won't help with type safety or anything, since you can enter anything you want into any provider extension

Reply:
(in general, with provider extensions, my aim is to mimic whatever they use in their api as closely as possible, to avoid confusion)

Reply:
i guess it's debatable here - I'll change it since we are on a slightly different level (image element vs within image_url)

Reply:
makes sense to have match the provider spec 1:1 for this, got it!